model/src/main/kotlin/config/PackageConfiguration.kt

```diff
@@ -80,34 +82,43 @@ data class PackageConfiguration(
         ) {
             "A package configuration must contain at most one of 'sourceArtifactUrl', 'vcs' or 'sourceCodeOrigin'."
         }
-    }
 
-    fun matches(otherId: Identifier, provenance: Provenance): Boolean {
-        @Suppress("ComplexCondition")
-        if (!id.type.equals(otherId.type, ignoreCase = true) ||
-            id.namespace != otherId.namespace ||
-            id.name != otherId.name ||
-            id.version != otherId.version
-        ) {
-            return false
+        if (id.isVersionRange()) {
+            require(
+                (vcs == null && sourceArtifactUrl == null)
```

Comment:
These checks could be combined like `require(!id.isVersionRange() || (vcs == null && sourceArtifactUrl == null))`.

Reply:
> Are you strongly opinionated on this ?

Not strongly, no, it's just a personal preference. However, if you keep the current version, the parentheses around `(vcs == null && sourceArtifactUrl == null)` should be dropped.